engine/collection/ingest/engine.go

```diff
@@ -224,6 +225,13 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 	// or a different cluster)
 	if originID == e.me.NodeID() {
 
+		// if the target cluster has no collection node other than possibly this
+		// collection node then don't attempt to propagate the transaction further
+		recipientIDs := txCluster.Filter(filter.Not(filter.HasNodeID(e.me.NodeID())))
+		if len(recipientIDs) == 0 {
+			return nil
```

Comment:
This should probably not return here, as that will miss the rest of the function, i.e. the info log statement. Maybe refactor the control flow